### PR TITLE
Jetpack Backup: add preflight checks on restore confirmation page

### PR DESCRIPTION
**Pull request by Initsogar**
Related to https://github.com/Automattic/jetpack-backup-team/issues/290

## Proposed Changes

* Enqueue and validate preflight checks on the restore confirmation page, only available under the `jetpack/backup-restore-preflight-checks` feature flag.
  * It will enqueue and validate preflight checks only when the site has no credentials configured, or they are invalid.
* Add a **Missing Credentials** message when a site starts a restore with no credentials and preflight checks fail.
  CleanShot 2024-02-01 at 00 18 35@2x
* Add Tracks events on `Enter credentials` and `Go back` button on the missing credentials message:
  * Enter credentials: `calypso_jetpack_backup_restore_missing_credentials_cta`
  * Go back: `calypso_jetpack_backup_restore_missing_credentials_back`

## Demo
Here's a quick demo of the new **Missing Credentials** page.

https://github.com/Automattic/wp-calypso/assets/1488641/5ab777d3-da0a-4b49-aedf-fe6d45593a75

## Testing Instructions
* Spin up a Calypso or Jetpack Cloud live branch
* Continue the flow depending on the scenario you want to test:

### Regression test
Given we are adding this functionality under the `jetpack/backup-restore-preflight-checks` feature flag, we need to ensure everything is still working as usual if the feature flag is disabled.

**> Site with credentials**
* Pick a site with a Jetpack VaultPress Backup plan and ensure it has working credentials configured.
* Navigate to Backup page.
* Pick a backup and click on `Restore to this point`
* Confirm the items you want to restore and click on `Confirm restore`
* The restore should complete and should display the `Your site has been successfully restored` message at the end.

**> Site without credentials (or non-working credentials)**
* Pick a site with a Jetpack VaultPress Backup plan and ensure it does not have credentials configured.
* Navigate to the Backup page.
* You should see the `Add your server credentials` banner on top and restore buttons should be disabled.

### New feature testing
Now that we ensured everything is working okay, we could test the new feature using the `jetpack/backup-restore-preflight-checks` flag.

* Pass `?flags=jetpack/backup-restore-preflight-checks` at the end of the URL of your Calypso/Jetpack Cloud live branch site.
* Pick a site with a Jetpack VaultPress Backup plan
* Navigate to the Backup page
* Ensure the `Add your server credentials` is not visible
* Ensure restore buttons are not disabled
* Pick a backup and click on `Restore to this point`
* Confirm the items you want to restore and click on `Confirm restore`
* Let's continue, depending on the scenario you would like to test:

**> Site with working credentials**
* The restore should complete and should display the `Your site has been successfully restored` message at the end.

**> Site with no credentials and preflight checks are OK**
* The restore will attempt to start and then an error will be thrown. This is because we are not allowing credential-less restores on WPCOM API yet (it validates if the site has credentials). This is OK, we need to revisit it and test it when backend is ready.
  CleanShot 2024-02-01 at 00 38 41@2x

**> Site with no credentials and preflight checks are failing**
* The restore will attempt to start and then it will display the **Missing Credentials** message.
  CleanShot 2024-02-01 at 00 41 08@2x
* Ensure that if you go back to the Backup page, you should see the `Add your server credentials` banner and restore buttons should be disabled.

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by Initsogar** [comment]
Investigating a bug that prevents start a restore when `GET
wpcom/v2/sites/ID/rewind/preflight` has previous preflight results.

File: client/my-sites/backup/rewind-flow/steps/missing-credenitals.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { useDispatch } from 'calypso/state';
+import { recordTracksEvent } from 'calypso/state/analytics/actions/record';
+
+interface Props {
+	siteSlug: string | null;
+	enterCredentialsEventName: string;
+	goBackEventName: string;
+}
+
+const MissingCredentials: FunctionComponent< Props > = ( {
```
Comment (Initsogar) [inline client/my-sites/backup/rewind-flow/steps/missing-credenitals.tsx]: We could start moving these flow steps into a separate components like this one.

File: client/my-sites/backup/rewind-flow/steps/missing-credenitals.tsx
Comment (jwebbdev) [inline client/my-sites/backup/rewind-flow/steps/missing-credenitals.tsx:1]: Typo in the filename `missing-credentials.tsx`

File: client/state/rewind/preflight/hooks.tsx [hunk 2]
```diff
@@ -29,8 +31,8 @@ export const usePreflightStatusQuery = ( siteId: number ): UseQueryResult => {
 				apiNamespace: 'wpcom/v2',
 			} );
 		},
-		enabled: shouldFetch(),
-		refetchInterval: 5000,
+		enabled: shouldFetch,
+		refetchInterval: 3000,
```
Comment (jwebbdev) [inline client/state/rewind/preflight/hooks.tsx]: Anything that led us to 3s from 5s?  Just curious if there was a technical reason or more of "it felt too slow"

Comment (Initsogar) [reply client/state/rewind/preflight/hooks.tsx]: I was messing around with my dev environment and forgot to keep it as it was. I think 5s still good. I just switched it back.

File: client/my-sites/backup/rewind-flow/steps/missing-credenitals.tsx
Comment (Initsogar) [reply client/my-sites/backup/rewind-flow/steps/missing-credenitals.tsx:1]: Fixed, thank you!

**Review by jwebbdev — APPROVED** [review]
LGTM

**Comment by Initsogar** [comment]
> LGTM

Thank you! I will wait for GH actions to E2E to proceed with the merge :)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11179460

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @Initsogar for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11179460).

**Pull request merged**